compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/BlendedManifest.java

```diff
@@ -69,8 +69,20 @@ public static BlendedManifest from(DependencyManifest dependencyManifest,
                     depInPkgManifest.repository().equals(ProjectConstants.LOCAL_REPOSITORY_NAME) ?
                     Repository.LOCAL : Repository.NOT_SPECIFIED;
 
-            if (!localPackageRepository.isPackageExists(depInPkgManifest.org(), depInPkgManifest.name(),
-                                                       depInPkgManifest.version())) {
+            if (depInPkgManifest.repository() != null &&
+                    depInPkgManifest.repository().equals(ProjectConstants.LOCAL_REPOSITORY_NAME)) {
```

Comment:
How does it behave when the value provided for `repo` in Ballerina.toml is something other than `"local"`?
The expected behavior is taht we should skip the dependency of PackageManifest and accept the dependency coming from DependencyManifest.

Reply:
If this is not handled, we can do it as follows:
1. Check for the value when building the `PackageManifest` in `ManifestBuilder` and report a diagnostic
2. Here, in the BlendedManifest simply add a `continue` if the repo is not `local`

Reply:
I updated this as per your specifications. Now if `repository` exists and its anything other than `local`, the PackageManifest is skipped and the dependency manifest is considered. 

However, if the local dependency does exist, that takes priority

Reply:
We should issue an error doagnostic at ManifestBuilder if the value is anything other than "local". Please check and add this if it is not added.

Reply:
Added an error diagnostic at the ManifestBuilder along with a test case to assert the diagnostic in cc388e8cb97c207da6be56ea2937cd626a376d67